src/snyk/base/services/authenticationService.ts

```diff
@@ -107,6 +108,17 @@ export class AuthenticationService implements IAuthenticationService {
     await this.configuration.setToken(token);
   }
 
+  async updateToken(token: string): Promise<void> {
+    if (!token) {
+      await this.initiateLogout();
+    } else {
+      if (!uuidValidate(token)) return Promise.reject(new Error('The entered token has an invalid format.'));
+
+      await this.initiateLogout();
```

Comment:
I don't think you need to call logout, because it will bring a user to the welcome view.